packages/amplify-cli-core/src/utils/packageManager.ts

```diff
@@ -63,37 +58,29 @@ export const getPackageManager = (rootPath?: string): PackageManager | null => {
     return null;
   }
 
+  // checks for yarn
   tempFilePath = path.join(effectiveRootPath, packageManagers.yarn.lockFile);
-
-  // checks for yarn2
-  if (packageManagers.yarn2.yarnrcPath !== undefined) {
-    const yarnRcFilePath = path.join(effectiveRootPath, packageManagers.yarn2.yarnrcPath);
-    if (fs.existsSync(tempFilePath) && checkExecutable(packageManagers.yarn.executable) && fs.existsSync(yarnRcFilePath)) {
-      return packageManagers.yarn2;
-    }
-  }
-
   if (fs.existsSync(tempFilePath) && checkExecutable(packageManagers.yarn.executable)) {
-    return packageManagers.yarn;
+    return await getYarnPackageManager(rootPath);
   }
 
+  // checks for npm
   tempFilePath = path.join(effectiveRootPath, packageManagers.npm.lockFile);
-
   if (fs.existsSync(tempFilePath)) {
     return packageManagers.npm;
   }
 
-  // No lock files present at this point
-  // check for yarn2
-  if (packageManagers.yarn2.yarnrcPath !== undefined) {
-    const yarnRcFilePath = path.join(effectiveRootPath, packageManagers.yarn2.yarnrcPath);
-    if (fs.existsSync(yarnRcFilePath) && checkExecutable(packageManagers.yarn.executable)) {
-      return packageManagers.yarn2;
-    }
-  }
+  // no lock files found
   if (checkExecutable(packageManagers.yarn.executable)) {
-    return packageManagers.yarn;
+    return await getYarnPackageManager(rootPath);
```

Comment:
The await here shouldn't be needed

```suggestion
    return getYarnPackageManager(rootPath);
```